sql/2019/10_SEO/10_09.sql

```diff
@@ -0,0 +1,15 @@
+#standardSQL
+
+# Content - looking at word count, thin pages, header usage, alt attributes images
+
+
+SELECT
+    url,
+    JSON_EXTRACT_SCALAR(REPLACE(JSON_EXTRACT_SCALAR(payload, '$._almanac'), '\\"','"'), "$.seo-words.wordsCount") as wordsCount,
```

Comment:
@rviscomi is there a way to get a statistical (normal) distribution of the wordCount?

Reply:
Yes, see `APPROX_QUANTILES` and `OFFSET`. Here's an example using a similar UDF to find the median:

```sql
CREATE TEMPORARY FUNCTION parseWordCount(payload STRING)
RETURNS NUMERIC LANGUAGE js AS '''
  try {
    var $ = JSON.parse(payload);
    var almanac = JSON.parse($._almanac);
    return almanac['seo-words'].wordsCount;
  } catch (e) {
    return 0;
  }
''';

SELECT
  APPROX_QUANTILES(parseWordCount(payload), 1000)[OFFSET(500)]
FROM
  `httparchive.almanac.pages_desktop_1k`
```

Reply:
Thanks. Added.